src/pytest_codspeed/instruments/valgrind.py

```diff
@@ -61,7 +53,8 @@ def measure(
         **kwargs: P.kwargs,
     ) -> T:
         self.benchmark_count += 1
-        if self.lib is None:  # Thus should_measure is False
+
+        if not self.should_measure:
```

Comment:
since the check is done here, we no longer need to check in the rest of the lib. It needs to be clear in the spec of the functions but it will help